perception/ground_segmentation/src/scan_ground_filter_nodelet.cpp

```diff
@@ -252,26 +251,37 @@ void ScanGroundFilterComponent::checkDiscontinuousGndGrid(
     abs(local_slope) < local_slope_max_angle_rad_ ||
     abs(tmp_delta_avg_z) < non_ground_height_threshold_ ||
     abs(tmp_delta_max_z) < non_ground_height_threshold_) {
-    gnd_cluster.addPoint(p.radius, p.orig_point->z);
     p.point_state = PointLabel::GROUND;
   } else if (local_slope > global_slope_max_angle_rad_) {
     p.point_state = PointLabel::NON_GROUND;
   }
 }
 
 void ScanGroundFilterComponent::checkBreakGndGrid(
-  PointRef & p, const std::vector<GridCenter> & gnd_grids_list, PointsCentroid & gnd_cluster)
+  PointRef & p, const std::vector<GridCenter> & gnd_grids_list)
 {
   float tmp_delta_avg_z = p.orig_point->z - gnd_grids_list.back().avg_height;
   float tmp_delta_radius = p.radius - gnd_grids_list.back().radius;
   float local_slope = std::atan(tmp_delta_avg_z / tmp_delta_radius);
   if (abs(local_slope) < global_slope_max_angle_rad_) {
-    gnd_cluster.addPoint(p.radius, p.orig_point->z);
     p.point_state = PointLabel::GROUND;
   } else if (local_slope > global_slope_max_angle_rad_) {
     p.point_state = PointLabel::NON_GROUND;
   }
 }
+void ScanGroundFilterComponent::recheckGroundCluster(
+  PointsCentroid & gnd_cluster, const float non_ground_threshold,
+  pcl::PointIndices & non_ground_indices)
+{
+  float min_gnd_height = gnd_cluster.getMinHeight();
+  pcl::PointIndices gnd_indices = gnd_cluster.getIndices();
+  std::vector<float> height_list = gnd_cluster.getHeightList();
+  for (size_t i = 0; i < height_list.size(); i++) {
```

Comment:
maybe better to use ++i for calc cost
```suggestion
  for (size_t i = 0; i < height_list.size(); ++i) {
```